src/components/Toggle/index.tsx

```diff
@@ -0,0 +1,210 @@
+    &[data-checked='true'] {
+      background: ${({ theme }) =>
+        theme.colors.primary.backgroundStrongDisabled};
+    }
+  }
+`
+
+const StyledCheckbox = styled.input`
+  position: absolute;
+  opacity: 0;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  cursor: pointer;
+
+  &[disabled] {
+    cursor: not-allowed;
+  }
+`
+
+const StyledLabel = styled.label<{
+  'data-size': 'small' | 'large'
+}>`
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+
+  &:active ${StyledToggle}[data-disabled='false']:after {
+    width: ${({ 'data-size': dataSize }) => SIZES[dataSize].ball * 1.3775}px;
+  }
+
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+    color: ${({ theme }) => theme.colors.neutral.textDisabled};
+  }
+`
+
+const StyledLabelContent = styled.span<{
+  labelPosition: 'left' | 'right'
+}>`
+  ${({ theme, labelPosition }) =>
+    labelPosition === 'left'
+      ? `margin-right: ${theme.space['1']}`
+      : `margin-left: ${theme.space['1']}`}
+`
+
+const LabelContent = ({
+  label,
+  labelPosition,
+}: Required<Pick<ToggleProps, 'label' | 'labelPosition'>>) => {
+  if (typeof label === 'string') {
+    return (
+      <StyledLabelContent labelPosition={labelPosition}>
+        {label}
+      </StyledLabelContent>
+    )
+  }
+
+  return <>{label}</>
+}
+
+type ToggleProps = {
+  id?: string
+  checked?: boolean
+  name: string
+  tooltip?: string
+  onChange?: ChangeEventHandler<HTMLInputElement>
+  size?: 'large' | 'small'
+  labelPosition?: 'left' | 'right'
+  label?: ReactNode
+  disabled?: boolean
+}
+
+const Toggle = ({
+  checked = false,
+  disabled = false,
+  id,
+  name,
+  onChange,
+  size = 'large',
+  tooltip,
+  labelPosition = 'right',
+  label,
+}: ToggleProps) => {
+  const [state, setState] = useState(checked)
+
+  const onLocalChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      onChange?.(event)
+      setState(event.target.checked)
+    },
+    [onChange, setState],
+  )
+
+  return (
+    <Tooltip text={tooltip}>
+      <StyledLabel
+        aria-disabled={disabled}
+        data-size={size}
+        onClick={evt => evt.stopPropagation()}
+      >
+        {label && labelPosition === 'left' ? (
+          <LabelContent label={label} labelPosition={labelPosition} />
+        ) : null}
+        <StyledToggle
+          data-size={size}
+          data-checked={state}
+          data-disabled={disabled}
+        >
+          <StyledCheckbox
+            id={id || name}
+            aria-label={name}
+            checked={state}
+            aria-checked={state}
+            disabled={disabled}
+            name={name}
+            onChange={onLocalChange}
+            type="checkbox"
+          />
+        </StyledToggle>
+        {label && labelPosition === 'right' ? (
+          <LabelContent label={label} labelPosition={labelPosition} />
+        ) : null}
+      </StyledLabel>
+    </Tooltip>
```

Comment:
There is no more label inside the toggle ? We use it in our console we will need to replace them

Reply:
@DorianMaliszewski there is no more, as seen with design the toggle is explicit enough to make the user understand that option is on / off. All toggle with text inside on console will be soon replaced with no text inside and a label on left or right side 😄